cityofaustin/atd-moped
# Normalize subcomponents schema

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/11452

This PR adds a many <> many table between `moped_components` and `moped_subcomponents`. We will be adding many more subcomponents in the near future, so it seemed worthwhile to get this taken care of.




## Testing
**URL to test:** Local

**Steps to test:**
1. Create new components with subcomponents. Signals of subtype PHB, Traffic, and RRFB  should have three subcomponent options: Audible push button, Bicycle signal, Leading pedestrian interval. Bike lane components (except **Ramp to Sidewalk** and **Advisory**) have Raised pavement markers.

2. Edit components by adding and removing subcomponents.

3. If you're feeling thorough, you can run this against a copy of production. I have done this and the data looks goood.

4. You can peep the components data dictionary (https://localhost:3000/moped/dev/lookups) —it should match staging:

Component ID | Component name | Component subtype | Geometry type | Subcomponents
-- | -- | -- | -- | --
5 | Bike Box |   | Point |  
7 | Bike Lane | Buffered | Line | Raised pavement markers
8 | Bike Lane | Climbing | Line | Raised pavement markers
9 | Bike Lane | Colored | Line | Raised pavement markers
10 | Bike Lane | Contraflow | Line | Raised pavement markers
11 | Bike Lane | Protected One-Way | Line | Raised pavement markers
12 | Bike Lane | Protected Two-Way | Line | Raised pavement markers
21 | Bike Lane | Turn Lane | Line | Raised pavement markers
22 | Bike Lane | Wide Curb Lane | Line | Raised pavement markers
6 | Bike Lane |   | Line | Raised pavement markers
20 | Bike Parking | Corral | Point |  
19 | Bike Parking |   | Point |  
16 | Signal | PHB | Point | Audible push button, Bicycle signal, Leading pedestrian interval
17 | Signal | RRFB | Point | Audible push button, Bicycle signal, Leading pedestrian interval
18 | Signal | Traffic | Point | Audible push button, Bicycle signal, Leading pedestrian interval


---
#### Ship list
